great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -575,62 +585,54 @@ def get_column_median(self, column):
 
-    def get_column_quantiles(self, column, quantiles, allow_relative_error=False):
-        selects = None
+    def get_column_quantiles(self, column: str, quantiles, allow_relative_error: bool = False):
         if self.engine.dialect.name.lower() == "mssql":
             # mssql requires over(), so we add an empty over() clause
-            selects = [
+            selects: List[WithinGroup] = [
                 sa.func.percentile_disc(quantile)
                 .within_group(sa.column(column).asc())
                 .over()
                 for quantile in quantiles
             ]
         elif self.engine.dialect.name.lower() == "bigquery":
             # BigQuery does not support "WITHIN", so we need a special case for it
-            selects = [
+            selects: List[WithinGroup] = [
                 sa.func.percentile_disc(sa.column(column), quantile).over()
                 for quantile in quantiles
             ]
         else:
-            try:
-                if isinstance(
-                    self.engine.dialect, sqlalchemy_redshift.dialect.RedshiftDialect
-                ):
-                    # Redshift does not have a percentile_disc method, but does support an approximate version
-                    if allow_relative_error is True:
-                        selects = [
-                            sa.text(
-                                ", ".join(
-                                    [
-                                        "approximate "
-                                        + str(
-                                            stmt.compile(
-                                                dialect=self.engine.dialect,
-                                                compile_kwargs={"literal_binds": True},
-                                            )
-                                        )
-                                        for stmt in selects
-                                    ]
-                                )
-                            )
-                        ]
-                    else:
-                        raise ValueError(
-                            "Redshift does not support computing quantiles without approximation error; "
-                            "set allow_relative_error to True to allow approximate quantiles."
-                        )
-            except (AttributeError, TypeError):
-                pass
-
-        if not selects:
-            selects = [
+            selects: List[WithinGroup] = [
                 sa.func.percentile_disc(quantile).within_group(sa.column(column).asc())
                 for quantile in quantiles
             ]
 
-        quantiles = self.engine.execute(
-            sa.select(selects).select_from(self._table)
-        ).fetchone()
+        try:
+            sql_engine_dialect: PGDialect_psycopg2 = self.engine.dialect
+            if isinstance(sql_engine_dialect, RedshiftDialect) or isinstance(sql_engine_dialect, PGDialect_psycopg2):
```

Comment:
This seems to me as though it would inappropriately attempt this codepath for users who are using psycopg2 with a true postgres (not redshift) database. Is that the case? I think we should actually verify whether or not the approximate and/or non-approximate methods are available, and that that should most likely happen during init rather than here.

More minor nit given the comments above, couldn't both of these `isinstance` checks be against `self.engine.dialect`?

Reply:
I will fix.  The main issue I was chasing is that the RedshiftDialect was not True in my situation.  I actually had the code before the way you are describing, but "shortened it for elegance".  Will bring back now.  In next commit.